src/ameba/rule/layout/trailing_blank_lines.cr

```diff
@@ -13,11 +13,18 @@ module Ameba::Rule::Layout
       description "Disallows trailing blank lines"
     end
 
-    MSG = "Blank lines detected at the end of the file"
+    MSG               = "Unnecessary blank line detected at the end of the file"
+    MSG_FINAL_NEWLINE = "Final newline missing"
 
     def test(source)
-      if source.lines.size > 1 && source.lines[-2, 2].join.strip.empty?
-        issue_for({source.lines.size - 1, 1}, MSG)
+      source_lines = source.lines
+      last_source_line = source_lines.last
+      source_lines_size = source.lines.size
+      return if source_lines_size == 1 && last_source_line.empty?
+
+      last_line_not_empty = !last_source_line.strip.empty?
```

Comment:
Calling `strip` here will remove excess whitespaces which misses the point of this rule.

Reply:
Disregard that, as you've noticed I mistakenly thought of _trailing whitespaces_ rule here.